File: src/Algorithm/LeetCode/LeetCode_674_Longest_Continuous_Increasing_Subsequence/Solution_DP.java
package Algorithm.LeetCode.LeetCode_674_Longest_Continuous_Increasing_Subsequence;

//https://www.programmercarl.com/0674.最长连续递增序列.html
class Solution_DP {
    public int findLengthOfLCIS(int[] nums) {
        int result = 1;
        int length = nums.length;
        int[] dp = new int[length];

        for (int i = 0; i < length; i++) {
            dp[i] = 1;
        }

        for (int i = 1; i < length; i++) {
            if (nums[i] > nums[i - 1]) {
                dp[i] = dp[i - 1] + 1;
            }

            result = Math.max(dp[i], result);
        }

        return result;
    }
}

File: src/Algorithm/LeetCode/LeetCode_647_Palindromic_Substrings/Solution_DP.py
# https://www.programmercarl.com/0647.回文子串.html
class Solution_DP:
    def countSubstrings(self, s: str) -> int:
        length = len(s)
        dp = [[False] * length for _ in range(length)]
        result = 0

        for i in range(length - 1, -1, -1):
            for j in range(i, length):
                if s[i] == s[j]:
                    if j - i <= 2:
                        dp[i][j] = True
                        result += 1
                    elif dp[i + 1][j - 1]:
                        dp[i][j] = True
                        result += 1

        return result
